src/tests/ftest/io/mdtest_small.py

```diff
@@ -44,6 +44,62 @@ def test_mdtest_small(self):
+        number_of_files_dirs = self.params.get("number_of_files_dirs",
+                                               "/run/mdtest/*")
+
+        # Running mdtest for different variants
+        for flag in mdtest_flags:
+            self.mdtest_cmd.flags.update(flag)
+            if self.mdtest_cmd.flags.value == ' ':
+                for api in mdtest_api:
+                    self.mdtest_cmd.api.update(api)
+                    if self.mdtest_cmd.api.value == 'DFS':
+                        self.mdtest_cmd.write_bytes.update(read_write[1])
+                        self.mdtest_cmd.read_bytes.update(read_write[1])
+                    else:
+                        self.mdtest_cmd.write_bytes.update(read_write[0])
+                        self.mdtest_cmd.read_bytes.update(read_write[0])
+                    for branch in branch_factor:
+                        self.mdtest_cmd.branching_factor.update(branch)
+                        if (self.mdtest_cmd.branching_factor.value ==
+                                branch_factor[0]):
+                            self.mdtest_cmd.num_of_files_dirs.update(
+                                number_of_files_dirs)
+                            for depth in dir_depth[:-1]:
+                                self.mdtest_cmd.depth.update(depth)
+                                self.execute_mdtest()
+                                self.mdtest_cmd.num_of_files_dirs.update(None)
+                        else:
+                            self.mdtest_cmd.items.update(num_of_items)
+                            self.mdtest_cmd.depth.update(dir_depth[2])
+                            self.execute_mdtest()
+                            self.mdtest_cmd.items.update(None)
+            else:
+                for api in mdtest_api:
+                    self.mdtest_cmd.api.update(api)
+                    if self.mdtest_cmd.api.value == 'POSIX':
+                        self.mdtest_cmd.write_bytes.update(read_write[1])
+                        self.mdtest_cmd.read_bytes.update(read_write[1])
+                        self.mdtest_cmd.branching_factor.update(
+                            branch_factor[0])
+                        self.mdtest_cmd.num_of_files_dirs.update(
+                            number_of_files_dirs)
+                        self.mdtest_cmd.depth.update(dir_depth[1])
+                    else:
+                        self.mdtest_cmd.write_bytes.update(read_write[0])
+                        self.mdtest_cmd.read_bytes.update(read_write[0])
+                        self.mdtest_cmd.branching_factor.update(
+                            branch_factor[1])
+                        self.mdtest_cmd.items.update(num_of_items)
+                        self.mdtest_cmd.depth.update(dir_depth[2])
+
+                    self.execute_mdtest()
+                    self.mdtest_cmd.num_of_files_dirs.update(None)
+                    self.mdtest_cmd.items.update(None)
```

Comment:
Can we add some comments to this big if # else and for loops? It might be difficult for someone to understand what is going on which this piece of code without putting some debug statements.